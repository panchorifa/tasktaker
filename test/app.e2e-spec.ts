import { Test, TestingModule } from '@nestjs/testing';
import { INestApplication } from '@nestjs/common';
import * as request from 'supertest';
import { AppModule } from './../src/app.module';

describe('AppController (e2e)', () => {
  let app: INestApplication;

  beforeEach(async () => {
    const moduleFixture: TestingModule = await Test.createTestingModule({
      imports: [AppModule],
    }).compile();

    app = moduleFixture.createNestApplication();
    await app.init();
  });

  it('/tasks (POST)', async () => {
    const requestBody = {
      priority: 'low',
      skills: ['skill1', 'skill2']};
    const res = await request(app.getHttpServer())
      .post('/tasks')
      .send(requestBody);
    expect(res.status).toBe(201);
  });

  it('/tasks/:id (PUT) 404 - not found', async () => {
    const requestBody = { completed: true }
    const res = await request(app.getHttpServer())
      .put('/tasks/1')
      .send(requestBody);
    expect(res.status).toBe(404);
  });

  it('/agents (GET)', async () => {
    const res = await request(app.getHttpServer()).get('/agents');
    expect(res.status).toBe(200);
    expect(res.body).toMatchSnapshot();
  });
});


// it('/tasks (GET)', async () => {
//   const res = await request(app.getHttpServer()).get('/tasks');
//   expect(res.status).toBe(200);
//   expect(res.body).toEqual([]);
// });
