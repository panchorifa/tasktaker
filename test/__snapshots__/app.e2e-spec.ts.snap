// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppController (e2e) /agents (GET) 1`] = `
Array [
  Object {
    "id": 1,
    "name": "one",
    "skills": Array [
      "skill1",
    ],
  },
  Object {
    "id": 2,
    "name": "two",
    "skills": Array [
      "skill1",
      "skill2",
    ],
  },
  Object {
    "id": 3,
    "name": "three",
    "skills": Array [
      "skill1",
      "skill2",
      "skill3",
    ],
  },
]
`;
